extensions/amp-google-document-embed/0.1/amp-google-document-embed.js

```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Embeds the Google Docs viewer
+ *
+ * Example:
+ * <code>
+ * <amp-google-document-embed
+ *   layout="fixed-height"
+ *   src="https://www.example.com/my-document.pdf"
+ *   height="600">
+ * </amp-google-document-embed>
+ * </code>
+ */
+
+import {addParamToUrl} from '../../../src/url';
+import {isExperimentOn} from '../../../src/experiments';
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {removeElement} from '../../../src/dom';
+import {user} from '../../../src/log';
+
+export const TAG = 'amp-google-document-embed';
+
+const ATTRIBUTES_TO_PROPAGATE = [
+  'alt',
+];
+
+const GOOGLE_DOCS_EMBED_RE = /^https?:\/\/docs\.google\.com.+\/pub.*\??/;
+
+export class AmpDriveViewer extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {?Element} */
+    this.iframe_ = null;
+  }
+
+  /**
+   * Prefetches and preconnects URLs related to the viewer.
+   * @param {boolean=} opt_onLayout
+   * @override
+   */
+  preconnectCallback(opt_onLayout) {
+    this.preconnect.url('https://docs.google.com', opt_onLayout);
+  }
+
+  /** @override */
+  renderOutsideViewport() {
```

Comment:
hmm, any particular reason for `false`? If it is a heavy component, maybe a lower number like 0.75 (which we do for YT)

Reply:
I based the extension off of other iframe extensions, like `amp-instagram`, which used `false`. I'll measure this against the youtube embed payload and see if `.75` makes sense here.

Reply:
youtube's embed loads about 500kb of resources, and this viewer loads about 400kb of resources, so a `.75` or close to it could work well.